### PR TITLE
CLOUDP-118231: Decryption by GCP KMS E2E tests

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes
E2E testing of logs decryption via GCP

_Jira ticket:_ CLOUDP-118231

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: e2e/cloud_manager/decryption_localkey_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 MongoDB Inc
```
Comment (fmenezes) [inline e2e/cloud_manager/decryption_localkey_test.go:1]: file renamed from `e2e/cloud_manager/decryption_test.go`

File: e2e/cloud_manager/decryption_test.go [hunk 2]
```diff
@@ -11,73 +11,74 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-//go:build e2e || (decrypt && (cloudmanager || om44 || om50))
 
 package cloud_manager_test
 
 import (
+	"bufio"
```
Comment (fmenezes) [inline e2e/cloud_manager/decryption_test.go:18]: file renamed from `e2e/logs_decrypt.go`

File: e2e/cloud_manager/decryption_localkey_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+//go:embed decryption/localKey/*
+var files embed.FS
+
+const LocalKeyTestsInputDir = "decryption/localKey"
+
+func TestDecrypt(t *testing.T) {
+	cliPath, err := e2e.Bin()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
```
Comment (andreaangiolillo) [inline e2e/cloud_manager/decryption_localkey_test.go:38]: req.NoError(err)
```suggestion
req.NoError(err)
```

File: e2e/cloud_manager/decryption_localkey_test.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+	tmp := t.TempDir()
+
+	keyFile := path.Join(tmp, "localKey")
+	err = dumpToTempFile(files, path.Join(LocalKeyTestsInputDir, "localKey"), keyFile)
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (andreaangiolillo) [inline e2e/cloud_manager/decryption_localkey_test.go:46]: req.NoError(err)
```suggestion
req.NoError(err)
```

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**